modules/launcher/src/main/java/org/mule/runtime/module/launcher/MuleContainer.java

```diff
@@ -217,6 +235,52 @@ protected void init(String[] args) throws IllegalArgumentException, Initialisati
     artifactResourcesRegistry.getMemoryManagementService().initialise();
     artifactResourcesRegistry.inject(artifactResourcesRegistry.getContainerProfilingService());
     initialiseIfNeeded(artifactResourcesRegistry.getContainerProfilingService());
+
+    if (isFipsEnabled()) {
+      configureSecurityManager();
+      setSecurityAlgorithm();
+    }
+  }
+
+  private void configureSecurityManager() throws InitialisationException {
+    try {
+      Class<?> classDef =
+          Class.forName(System.getProperty(PROPERTY_FIPS_PROVIDER, "org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider"));
+      Constructor<?> constructor = classDef.getConstructor();
+      Provider fipsProvider = (Provider) constructor.newInstance();
+      java.security.Security.insertProviderAt(fipsProvider, 1);
```

Comment:
do we really need to use the fully qualified name for the `Security` class here? what is it colliding with? Actually we should be using a static import here to follow code conventions.

Reply:
Its safer to specify full name here, It will be instantiated later by JVM as needed.  From docs
```
public static [Class](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html)<?> forName([String](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html) className)
                        throws [ClassNotFoundException] 

Parameters:
className - the fully qualified name of the desired class.
```

Reply:
sorry, the lines marked for my comment are confusing, but I was referring to:

```suggestion
      Security.insertProviderAt(fipsProvider, 1);
```
or even:
```suggestion
      insertProviderAt(fipsProvider, 1);
```

Reply:
I think Security.insertProviderAt is bit better as that way we know its in a different class

Reply:
Personally, I also prefer `Security.insertProviderAt`, but our [code conventions](https://docs.google.com/document/d/1N2Axq3ABJjYmj9BjfYQkvJfF0odHQnlb8MAdVQydOWI/edit#heading=h.ujxk1qbe2hxj) say this:

> Static imports
> Always use static imports unless it’s not clear what the method is doing without using the class name when referencing the method/variable.

This one is kind of greyish, because `insertProviderAt` does not indicate what is the provider for.

Each reviewer may have their own opinion about when is it clear and when is it not.

For this particular case, I will approve either option.

Reply:
Im not in favor of that as you can imagine you could have static imports from say 5 classes and can get very confusing. But I guess following agreed upon conventions is important.